KreativeName1/RunTrack
# Improve UI Responsiveness by Offloading Password Validation to Background Task

### Overview
This pull request enhances the responsiveness of the UI during password validation by executing the `ValidatePassword()` method in a background task. 

### Changes Made
- The password validation process, which can take between 223 and 589 milliseconds due to the BCrypt.Net.BCrypt.Verify method, has been moved to a background task using `Task.Run()`.
- Access to UI components within `ValidatePassword()` is ensured to occur on the UI thread via `Application.Current.Dispatcher.Invoke()`, preventing potential threading issues.

### Benefits
- Users will experience a smoother interface without delays when validating passwords, improving overall usability.
- The changes ensure that long-running operations do not block the UI thread.

### How to Test
1. Run the application.
2. Trigger the password validation process and observe that the UI remains responsive during the validation period.
3. Verify that password validation results are displayed correctly without any delays.

### Related Issues
- N/A
